File: client/src/app/_dtos/process/candidateFlightGrpDto.ts

export interface ICandidateFlightGrpDto
{
    id: number;
    dateOfFlight: Date;
    orderNo: number;

    airlineName: string;
    flightNo: string;
    airportOfBoarding: string;
    airportOfDestination: string;
    eTD_Boarding: Date;
    eTA_Destination: Date;
    airportVia: string;
    flightNoVia: string;
    eTA_Via?: Date | null;
    eTD_Via?: Date | null;
    countOfPax: number;
}

export class CandidateFlightGrpDto implements ICandidateFlightGrpDto
{
    id=0;
    dateOfFlight=new Date;
    orderNo=0;
    
    airlineName = '';   
    flightNo = ''; 
    airportOfBoarding = '';
    flightNoVia = ''; 
    airportOfDestination = '';
    eTD_Boarding= new Date;
    eTA_Destination= new Date;
    airportVia = '';
    eTD_Via= null;    //new Date;
    eTA_Via= null;  // new Date;
    countOfPax: number=0;
}

File: api/DTOs/Process/CandidateFlightGrpDto.cs
namespace api.DTOs.Process
{
    public class CandidateFlightGrpDto
    {
        public int Id { get; set; }
        public DateTime DateOfFlight { get; set; }
        public int OrderNo { get; set; }
        public string AirlineName { get; set; }
        public string FlightNo { get; set; }
        public string AirportOfBoarding { get; set; }
        public string AirportOfDestination { get; set; }
        public DateTime ETD_Boarding { get; set; }
        public DateTime ETA_Destination { get; set; }
        public string AirportVia { get; set; }
        public string FightNoVia { get; set; }
        public DateTime? ETA_Via { get; set; }
        public DateTime? ETD_Via { get; set; }
        public int CountOfPax { get; set; }
    }
}